internal/providers/testing/provider_mock.go

```diff
@@ -179,9 +180,21 @@ func (p *MockProvider) getResourceIdentitySchemas() providers.GetResourceIdentit
 		return *p.GetResourceIdentitySchemasResponse
 	}
 
-	return providers.GetResourceIdentitySchemasResponse{
-		IdentityTypes: map[string]providers.IdentitySchema{},
+	resp := providers.GetResourceIdentitySchemasResponse{IdentityTypes: make(map[string]providers.IdentitySchema)}
+	if p.GetProviderSchemaResponse != nil {
+
+		for typeName, schema := range p.GetProviderSchemaResponse.ResourceTypes {
+			if schema.Identity != nil {
```

Comment:
Usually the resource identity schemas come from the `GetResourceIdentitySchemas` call and not the other way around. I feel like extracting it from the ResourceTypes could lead to wrong assumptions

Reply:
Also, this is only a fallback when you do not provide a separate resource identity response.